kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -310,10 +310,8 @@ public Subscription createSubscription(long callingNumber, Language language, Ci
             subscriberDataService.create(subscriber);
         }
 
-        // todo: switch to subscriber.getLanguage() & .getCircleName() when the ticket is fixed
-        // https://applab.atlassian.net/browse/MOTECH-1678
-        Language subscriberLanguage = (Language) subscriberDataService.getDetachedField(subscriber, "language");
-        Circle subscriberCircle = (Circle) subscriberDataService.getDetachedField(subscriber, "circle");
+        Language subscriberLanguage = subscriber.getLanguage();
```

Comment:
MOTECH-1678 isn't fixed but you did the change anyway, I'm assuming it's because we put everything in transactions?